src/pages/iou/IOUModal.js

```diff
@@ -0,0 +1,173 @@
+import React, {Component} from 'react';
+import {View, TouchableOpacity} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import ONYXKEYS from '../../ONYXKEYS';
+import ROUTES from '../../ROUTES';
+import {redirectToLastReport} from '../../libs/actions/App';
+import IOUAmountPage from './steps/IOUAmountPage';
+import IOUParticipantsPage from './steps/IOUParticipantsPage';
+import IOUConfirmPage from './steps/IOUConfirmPage';
+import Header from '../../components/Header';
+import styles from '../../styles/styles';
+import Icon from '../../components/Icon';
+import getPreferredCurrency from '../../libs/actions/IOU';
+import {Close, BackArrow} from '../../components/Icon/Expensicons';
+
+/**
+ * IOU modal for requesting money and splitting bills.
+ */
+const propTypes = {
+
+    /* Onyx Props */
+    // Url currently in view
+    currentURL: PropTypes.string,
+
+    // IOU modal data
+    iouData: PropTypes.shape({
+        isLoadingCurrency: PropTypes.bool.isRequired,
+        isLoadingParticipants: PropTypes.bool.isRequired,
+        createReportInProgress: PropTypes.bool.isRequired,
+    }),
+};
+
+const StepType = {
+    IOUAmount: 'IOUAmount',
+    IOUParticipants: 'IOUParticipants',
+    IOUConfirm: 'IOUConfirm',
+};
+
+const defaultProps = {
+    currentURL: '',
+    iouData: {
+        isLoadingCurrency: true,
+        isLoadingParticipants: true,
+        createReportInProgress: false,
+    },
+};
+
+class IOUModal extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            steps: [StepType.IOUAmount, StepType.IOUParticipants, StepType.IOUConfirm],
+            currentStepIndex: 0,
+            hasMultipleParticipants: this.props.currentURL === ROUTES.IOU_BILL_SPLIT,
+        };
+
+        this.getTitleForStep = this.getTitleForStep.bind(this);
+        this.navigateToPreviousStep = this.navigateToPreviousStep.bind(this);
+        this.navigateToNextStep = this.navigateToNextStep.bind(this);
+    }
+
+    componentDidMount() {
+        getPreferredCurrency();
+    }
+
+    /**
+     * Returns the title for the currently selected page
+     *
+     * @return {String}
+     */
+    getTitleForStep() {
+        switch (this.state.steps[this.state.currentStepIndex]) {
```

Comment:
I would instead move this data into a constant at the top of the file and access it like:

```
return stepTitles[this.state.currentStepIndex
] || '';

It makes this function super simple.
```

Reply:
I know this will possible change how the steps are stored as well, but I think it might be good because this looks confusing:

```
const StepType = {
    IOUAmount: 'IOUAmount',
    IOUParticipants: 'IOUParticipants',
    IOUConfirm: 'IOUConfirm',
};
```

I would consider something like this to kill two birds with one stone if it's possible:
```
const steps  = {
    IOUAmount: 'Amount',
    IOUParticipants: 'Participants',
    IOUConfirm: 'Confirm',
};
```

Reply:
Heh I was going to suggest this, but in the end did not 😄

Reply:
> I would consider something like this to kill two birds with one stone if it's possible:

Nice improvement, yup this works perfectly.